include/dlaf/matrix/matrix.h

```diff
@@ -88,7 +101,18 @@ class Matrix : public Matrix<const T, D> {
   ///            of the local part of the matrix will be stored in memory,
   /// @pre distribution.localSize() == layout.size(),
   /// @pre distribution.blockSize() == layout.blockSize().
-  Matrix(Distribution distribution, const LayoutInfo& layout) noexcept;
+  Matrix(Distribution distribution, const LayoutInfo& layout) noexcept
```

Comment:
Same question here about `noexcept`, except that this already was `noexcept`...

Reply:
Removed `noexcept` from all the allocating constructors.